packages/story-editor/src/elements/text/karma/edit.karma.js

```diff
@@ -168,10 +168,17 @@ describe('TextEdit integration', () => {
           expect(fixture.querySelector('[data-testid="textEditor"]')).toBeNull()
         );
         // The element is still selected and updated.
-        const storyContext = await fixture.renderHook(() => useStory());
-        expect(storyContext.state.selectedElements[0].content).toEqual(
-          'This is some test text.'
-        );
+        await waitFor(async () => {
+          const story = await fixture.renderHook(() => useStory());
+
+          if (!story.state.selectedElements.length) {
+            throw new Error('story not ready');
+          }
+
+          expect(story.state.selectedElements[0].content).toEqual(
+            'This is some test text.'
+          );
+        });
```

Comment:
There's a `await fixture.events.sleep(100);` before this. It seems like we may not need to find the element with the id of `textEditor` either. All this test is trying to figure out is if we updated the text element in the story.

💭 may not need that since we're going to wait for the story to update.

```js
        // Exit edit mode using the Esc key
        await fixture.events.keyboard.press('Esc');

        await waitFor(async () => {
          const story = await fixture.renderHook(() => useStory());

          if (!story.state.selectedElements.length) {
            throw new Error('story not ready');
          }

          expect(story.state.selectedElements[0].content).toEqual(
            'This is some test text.'
          );
        });
```

Reply:
Good catch. Removed.